src/green/builder.rs

```diff
@@ -216,9 +216,9 @@ impl Builder {
 impl Builder {
     fn collect_root_nodes(&mut self) -> Vec<Arc<Node>> {
         // NB: `drain_filter` is `retain` but with an iterator of the removed elements.
```

Comment:
This line of the comment is wrong now that the parity is flipped to be determining the elements drained, not the elements retained.

I think the correct analogue would now be

```suggestion
        // NB: `drain_filter` is `iter().filter` but also removing the elements chosen.
```